### PR TITLE
refactor: mappable fields with static values version 2

### PR DESCRIPTION
**Pull request by benlife5**
Demo: https://www.loom.com/share/79a23c05d0984ced80d6bddc9bc0fe1e?sid=eb855a66-4130-4034-a3a2-ed445235cbe6

### CODE REVIEW

File: test/resolveYextEntityField.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      {
+        address: {},
+      },
+      { fieldName: "address.city", staticValue: "City" }
+    ),
+    "City"
+  );
+});
+
+test("resolveProp returns static value Use Static Value is selected", async () => {
```
Comment (asanehisa) [inline test/resolveYextEntityField.test.ts:44]: when Use Static Value is selected?

**Review by jwartofsky-yext — COMMENTED** [review]
lgtm for the most part

Could you include a video?

File: test/resolveYextEntityField.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { test, assert } from "vitest";
+import { resolveYextEntityField } from "../src/utils/resolveYextEntityField.ts";
+
+test("resolveProp returns value when field found in document", async () => {
```
Comment (jwartofsky-yext) [inline test/resolveYextEntityField.test.ts:4]: it's not called resolveProp anymore

File: test/resolveYextEntityField.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          city: "potato land",
+        },
+      },
+      { fieldName: "address.city", staticValue: "" }
+    ),
+    "potato land"
+  );
+});
+
+test("resolveProp returns empty value when field found in document", async () => {
```
Comment (jwartofsky-yext) [inline test/resolveYextEntityField.test.ts:18]: same

File: test/resolveYextEntityField.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      {
+        address: {},
+      },
+      { fieldName: "address.city", staticValue: "City" }
+    ),
+    "City"
+  );
+});
+
+test("resolveProp returns static value when 'Use Static Value' is selected", async () => {
```
Comment (jwartofsky-yext) [inline test/resolveYextEntityField.test.ts:44]: same

File: src/utils/resolveYextEntityField.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { EntityFieldType } from "../components/YextEntityFieldSelector.tsx";
+
+export const resolveYextEntityField = <T>(
+  document: any,
+  entityField: EntityFieldType
+): T => {
+  // return static value if "Use Static Value" selected
+  if (entityField.fieldName === "") {
```
Comment (jwartofsky-yext) [inline src/utils/resolveYextEntityField.ts:8]: Does the user have to unset the fieldName to use the static value?

Is this what we want?

Comment (jwartofsky-yext) [reply src/utils/resolveYextEntityField.ts:8]: If both staticValue and fieldName are set, the fieldName will be chosen.

Is this what we want?

File: src/components/YextEntityFieldSelector.tsx [hunk 6]
```diff
@@ -68,50 +45,27 @@ export const YextEntityFieldSelector = <T extends Record<string, any>>(
                   }),
                 ],
               }}
-              onChange={(value) => {
+              onChange={(selectedEntityFieldName) => {
                 onChange({
-                  name: value as unknown as string, // hack because the option value is a string so it comes back as a string even though TS thinks it's an object
-                  value: resolveProp(document, value as unknown as string),
+                  fieldName: selectedEntityFieldName as unknown as string, // hack because the option value is a string so it comes back as a string even though TS thinks it's an object
```
[the remark this answers is not in the document]
Comment (benlife5) [reply src/components/YextEntityFieldSelector.tsx]: was able to remove it without an issue

File: src/utils/resolveYextEntityField.ts (hunk 5 above)
Comment (mkilpatrick) [reply src/utils/resolveYextEntityField.ts:8]: Yes. And if it's not in the document it still falls back to staticValue.

File: src/utils/resolveYextEntityField.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    if (!missedStep) {
+      return current;
+    }
+  } catch (e) {
+    console.error("Error in resolveYextEntityField:", e);
+  }
+
+  // if field not found, return static value as a fallback
+  console.warn(
```
[the remark this answers is not in the document]
Comment (benlife5) [reply src/utils/resolveYextEntityField.ts:33]: The warning in getFilteredEntityFields is about the filter allowList including fields that never exist, this warning is about a selected field not existing in the document.

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**